app/assets/javascripts/home.coffee

```diff
@@ -26,19 +26,46 @@ set_coords_by_geolocation = (position) ->
   set_coords(position.coords.accuracy, position.coords.latitude, position.coords.longitude)
 
 block_callback = (err) ->
-  location_error()
+  if $('#location_geolocation').prop('checked')
+    $('#location_button').prop('innerHTML', 'Get My Location')
+
+  $('#error-geolocation').modal('hide')
+  $('#error-position_unavailable').modal('hide')
+
+  if err.code == err.POSITION_UNAVAILABLE && is_safari()
+    $('#error-position_unavailable').modal('show')
+  else 
+    $('#error-geolocation').modal('show')
 
   Sentry.setExtra("error_code", err.code)
   Sentry.setExtra("error_message", err.message)
   Sentry.captureException(err)
 
+is_safari = ->
+  ua = navigator.userAgent.toLowerCase()
+  if ua.indexOf('safari') != -1
+    if ua.indexOf('chrome') > -1
+      return false
+    return true
+  return false
+
+  
 get_location = ->
   if navigator.geolocation
     location_start()
     navigator.geolocation.getCurrentPosition set_coords_by_geolocation, block_callback
   else
-    # TODO replace with error specific for geolocation
     location_error()
+    if $('#location_geolocation').prop('checked')
+        $('#location_button').prop('innerHTML', 'Get My Location')
+      $('#error-geolocation').modal('show')
+
+ajax_interactions = ->
+  $(document)
+    .ajaxStart ->
+      location_start()
+    .ajaxStop ->
+      location_finished()
```

Comment:
What is being accomplished here? This `ajax_interactions()` function is defined, but I don't see it called anywhere. I'm also not certain it's a good idea to register global handlers [for *all* AJAX requests](https://api.jquery.com/ajaxStart/), since not all of them are related to fetching location. It would be better to call the necessary functions directly from the success and fail callbacks of the specific AJAX request that they relate to.

Reply:
It's not new code, I'm not sure why it's showing up as new. But that a good question, one that I should look into, not Jim.

Reply:
Whoops, thanks for catching that weird display thing, @ryanrolds! Sorry about my confusion, @JCoulter85! I'll resolve this conversation now, since it doesn't actually relate to this PR.

Since I thought it was new code, I only looked at the other new code. The function actually *is* called (about 30 lines down) when the DOM loads on the homepage.